### PR TITLE
Closes #3445 Make config dependencies, module dependencies in az_paragraphs:az_paragraphs_view

### LINKED ISSUE #3445: Add dependencies to Quickstart Paragraphs - Views Side-by-Side.
## Problem/Motivation
When attempting to install this module on a site without az_news installed produces an error.

### Describe the bug
Dependent node type not installed.

### To Reproduce

On a quickstart site disable az_news and delete all of its config, or create a site without news in the first place.

Then try to enable the Quickstart Paragraphs -  Views Side-by-Side module
See error

## Proposed resolution
Add dependencies needed for the view to work
https://github.com/az-digital/az_quickstart/blob/main/modules/custom/az_paragraphs/az_paragraphs_views_side_by_side/config/install/views.view.az_side_by_side_views.yml

## Alternate resolution
Put the view config file in the optional folder.

##  Screenshots
image

### PR DESCRIPTION
**Pull request by trackleft**
## Description

Adding config dependencies for az_paragraphs_view
https://github.com/az-digital/az_quickstart/blob/main/modules/custom/az_paragraphs/az_paragraphs_views_side_by_side/config/install/views.view.az_side_by_side_views.yml#L6-L9

Adding update function to install dependencies if the az_paragraphs_views_side_by_side module is already installed.
Adding the dependencies to the info file.

## Related issues
Closes #3445 

## How to test

On a quickstart site disable az_news and delete all of its config, or create a site without news in the first place.
Then try to enable the Quickstart Paragraphs - Views Side-by-Side module
See that you will need to install dependencies if you want to install Views Side-by-Side.

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [x] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by bberndt-uaz** [comment]
I created a local Quickstart site without installing the Quickstart News module. When I tried to install the Quickstart Paragraphs - Views Side-by-Side module, the prompt to confirm the installation of Quickstart News was displayed:

image

#####
However, since Quickstart SEO was installed, Quickstart News could not be installed.

image

#####
Once I reinstalled Quickstart without both Quickstart News and Quickstart SEO, I was able to install Quickstart Paragraphs - Views Side-by-Side successfully.

image

#####
I'm not sure what the best solution would be in this case.

**Review by RyanDool — DISMISSED** [review]

**Comment by trackleft** [comment]
Looks like we need another ticket for adding dependencies to  https://github.com/az-digital/az_quickstart/blob/main/modules/custom/az_seo/az_seo.info.yml

**Review by tadean — APPROVED** [review]

**Review by joeparsons — APPROVED** [review]

**Review by bberndt-uaz — APPROVED** [review]

**Comment by joeparsons** [comment]
Going to move this to the 2.10.6 project so we can do more testing / ensure that DB updates aren't needed.

**Review by tadean — DISMISSED** [review]

**Comment by danahertzberg** [comment]
Check for DB updates needed/not needed. @joeparsons to answer before merging.

File: modules/custom/az_paragraphs/az_paragraphs_views_side_by_side/az_paragraphs_views_side_by_side.install [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+/**
+ * Install az_news, az_person, az_event, az_flexible_page modules.
+ *
+ * Install modules that contain configuration dependencies.
+ *
+ * @see https://github.com/az-digital/az_quickstart/issues/3445
+ * @see https://github.com/az-digital/az_quickstart/pull/3446
+ */
+function az_paragraphs_views_side_by_side_update_1021101() {
```
Comment (joeparsons) [inline modules/custom/az_paragraphs/az_paragraphs_views_side_by_side/az_paragraphs_views_side_by_side.install:16]: If we want this to be an update for 2.10.x
```suggestion
function az_paragraphs_views_side_by_side_update_1021001() {
```

**Review by joeparsons — APPROVED** [review]

**Review by tadean — APPROVED** [review]

**Review by bberndt-uaz — APPROVED** [review]

**Pull request merged**